main/ui/src/main/java/org/cryptomator/ui/health/CheckController.java

```diff
@@ -32,52 +36,65 @@
 	private final ReportWriter reportWriter;
 	private final ExecutorService executorService;
 	private final ObjectProperty<HealthCheckTask> selectedTask;
+	private final ResourceBundle resourceBundle;
 	private final SimpleObjectProperty<Worker<?>> runningTask;
 	private final Binding<Boolean> running;
 	private final Binding<Boolean> finished;
 	private final BooleanBinding anyCheckSelected;
 	private final BooleanBinding readyToRun;
+	private final StringProperty runButtonDescription;
 
 	/* FXML */
 	public ListView<HealthCheckTask> checksListView;
 
 
 	@Inject
-	public CheckController(Lazy<Collection<HealthCheckTask>> tasks, ReportWriter reportWriteTask, ObjectProperty<HealthCheckTask> selectedTask, ExecutorService executorService) {
+	public CheckController(Lazy<Collection<HealthCheckTask>> tasks, ReportWriter reportWriteTask, ObjectProperty<HealthCheckTask> selectedTask, ExecutorService executorService, ResourceBundle resourceBundle) {
 		this.tasks = FXCollections.observableArrayList(tasks.get());
 		this.reportWriter = reportWriteTask;
 		this.executorService = executorService;
 		this.selectedTask = selectedTask;
+		this.resourceBundle = resourceBundle;
 		this.runningTask = new SimpleObjectProperty<>();
 		this.running = EasyBind.wrapNullable(runningTask).mapObservable(Worker::runningProperty).orElse(false);
 		this.finished = EasyBind.wrapNullable(runningTask).mapObservable(Worker::stateProperty).map(endStates::contains).orElse(false);
 		this.readyToRun = runningTask.isNull();
 		this.anyCheckSelected = selectedTask.isNotNull();
+		this.runButtonDescription = new SimpleStringProperty(resourceBundle.getString("health.check.runAllButton"));
 	}
 
 	@FXML
 	public void initialize() {
 		checksListView.setItems(tasks);
 		checksListView.setCellFactory(ignored -> new CheckListCell());
 		selectedTask.bind(checksListView.getSelectionModel().selectedItemProperty());
+		selectedTask.addListener(this::updateRunButtonDescription);
+	}
+
+	private void updateRunButtonDescription(ObservableValue<? extends HealthCheckTask> observable, HealthCheckTask oldTask, HealthCheckTask newTask) {
+		if (newTask == null) {
+			runButtonDescription.set(resourceBundle.getString("health.check.runSingleButton"));
+		} else if (oldTask == null && newTask != null) {
+			runButtonDescription.set(resourceBundle.getString("health.check.runAllButton"));
+		}
```

Comment:
This seems to be leftover from experimenting with using a single button and changing its description.

(Which is a good demonstration why having separate buttons is less complex)

Reply:
Fixed in 20f88e2